src/components/TodoList/TodoList.jsx

```diff
@@ -1,44 +1,111 @@
-import React from 'react';
+import React, { Component } from 'react';
+import cn from 'classnames';
+import PropTypes from 'prop-types';
+
 import './TodoList.scss';
 
-export const TodoList = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+export class TodoList extends Component {
+  state = {
+    postId: 0,
+    select: 'all',
+    filterTitle: '',
+  };
+
+  buttonClickHandler = (postId, userId) => {
+    this.setState({ postId });
+
+    this.props.onSelectUser(userId);
+  }
+
+  selectChangeHandler = (event) => {
+    this.setState({ select: event.target.value });
+  }
+
+  onInputHandler = (event) => {
+    this.setState({ filterTitle: event.target.value });
+  }
+
+  render() {
+    let { todos } = this.props;
+    const { postId, filterTitle, select } = this.state;
+
+    if (select !== 'all') {
+      select === 'active'
+        ? todos = todos.filter(todo => todo.completed === false)
+        : todos = todos.filter(todo => todo.completed === true);
+    }
+
+    if (filterTitle) {
+      todos = todos
+        .filter(({ title }) => title !== null
+          && title.toLowerCase().includes(filterTitle.toLowerCase()));
+    }
```

Comment:
let's save it to constant or create a separate function (watch the video)

`getFilteredTodos(todos)`

https://www.youtube.com/watch?v=RXVPOLx5C4A&feature=youtu.be